backend/typescript/services/implementations/taskService.ts

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { task, category, resident, staff } from "@prisma/client";
+import prisma from "../../prisma";
+import { ITaskService, TaskDTO, InputTaskDTO } from "../interfaces/taskService";
+import logger from "../../utilities/logger";
+import { getErrorMessage } from "../../utilities/errorUtils";
+
+const Logger = logger(__filename);
+
+type TaskExtended =
+  | null
+  | (task & {
+      category?: category | null;
+      assignee?: resident | null;
+      assigner?: staff | null;
+    });
+
+const convertTaskRelation = (taskExtended: TaskExtended): TaskDTO => {
+  if (!taskExtended) {
+    throw new Error("null taskRelation passed to convertTaskRelation");
+  }
+  const { category: theCategory, assignee, assigner } = taskExtended;
+  const category_name = theCategory == null ? undefined : theCategory.name;
+  const assignee_name =
+    assignee == null ? undefined : `${assignee.firstName} ${assignee.lastName}`;
+  const assigner_name =
+    assigner == null ? undefined : `${assigner.firstName} ${assigner.lastName}`;
+
+  const taskDTO: TaskDTO = {
+    ...taskExtended,
+    category_name,
+    assignee_name,
+    assigner_name,
+  };
+
+  return taskDTO;
+};
+
+class TaskService implements ITaskService {
```

Comment:
I tested all the API's and only delete is actually working. createTask worked when Nandini and I fixed it but now it needs fixing again. TODO: fix all APIs (Saturday?)